hat/assets/js/apps/Iaso/redux/useInjectedStore.ts

```diff
@@ -0,0 +1,63 @@
+import { useContext, useEffect, useMemo, useState } from 'react';
+import { ReactReduxContext } from 'react-redux';
+/**
+ *
+ * @deprecated Redux
+ * This is adding a list of reducers to the existing one's.
+ * As using redux is a practice we are trying to avoid, try not to use this hook
+ * Also do not use same reducer key as the existing one's (see hat/assets/js/apps/Iaso/redux/store.js)
+ *
+ * Example how to use in a component:
+ *  const store = useInjectedStore([
+        {
+            reducerKey: 'reducerKey',
+            reducer: IMPORTED_REDUCER,
+        },
+    ]);
+    if (!store) return null;
+    return (
+        <Provider store={store}>
+            <CUSTOM_COMPONENT />
+        </Provider>
+    );
+ *
+ */
```

Comment:
This makes me think we could go further than the hook and provide a custom `Provider` where the reducers are passed as props. Not blocking though, especially given the intended use:
```typescript
export const CombinedStoreProvider = ({reducers: Reducer[], children:ReactNode}) => {
    const store = useInjectedStore(reducers);
    if (!store) return null;
    return (
        <Provider store={store}>
           {children}
        </Provider>
    );
}

```